types/logger.d.ts

```diff
@@ -0,0 +1,68 @@
+import * as http from 'http'
+import * as net from 'net'
+
+import { FastifyError } from './error'
+import { RawServerBase, RawServerDefault, RawRequestBase, RawRequestDefault, RawReplyBase, RawReplyDefault } from './utils';
+
+/**
+ * Standard Fastify logging function
+ */
+export interface FastifyLogFn {
+  (msg: string, ...args: any[]): void;
+  (obj: object, msg?: string, ...args: any[]): void;
+}
+
+export type LogLevels = 'info' | 'error' | 'debug' | 'fatal' | 'warn' | 'trace'
+
+/**
+ * Fastify Custom Logger options.
+ */
+export interface FastifyLoggerOptions<
+  RawServer extends RawServerBase = RawServerDefault,
+  RawRequest extends RawRequestBase = RawRequestDefault<RawServer>,
+  RawReply extends RawReplyBase = RawReplyDefault<RawServer>
+> {
+  serializers?: {
+    req: (req: RawRequest) => {
+      method: string,
+      url: string,
+      version: string,
+      hostname: string,
+      remoteAddress: string,
+      remotePort: number
+    },
+    err: (err: FastifyError) => {
+      type: string;
+      message: string;
+      stack: string;
+      [key: string]: any;
+    },
+    res: (res: RawReply) => {
+      statusCode: string | number
+    }
+  };
+  info: FastifyLogFn;
+  error: FastifyLogFn;
+  debug: FastifyLogFn;
+  fatal: FastifyLogFn;
+  warn: FastifyLogFn;
+  trace: FastifyLogFn;
+  child: FastifyLogFn | FastifyLoggerOptions<RawServer>;
+  genReqId?: string;
+}
+
+/**
+Planning on using Pino as your logger?
+1. Download and install pino types `npm i -s @types/pino
+2. Add the following code block to a type declaration file
+```typescript
+import * as pino from 'pino' 
+namespace fastify {
+  interface FastifyInterface {
```

Comment:
Has `FastifyInterface` changed to `FastifyInstance`?

Reply:
Oh darn this information is out of date... fixing!

Reply:
<img width="809" alt="Screen Shot 2019-08-11 at 11 17 02 PM" src="https://user-images.githubusercontent.com/16144158/62844268-2a25a980-bc8e-11e9-8548-e0a621e14cdf.png">
I don't love this solution. I wish I could have use module declaration merging, but I couldn't figure that out so instead I've added a 4th generic parameter (totally optional) that enables the user to pass their own log types to the Fastify instance.